client/client_test.go

```diff
@@ -781,6 +782,178 @@ func TestProgressiveCall(t *testing.T) {
 	r.Close()
 }
 
+func TestProgressiveCalls(t *testing.T) {
+	// Connect two clients to the same server
+	callee, caller, r, err := connectedTestClients()
+	if err != nil {
+		t.Fatal("failed to connect test clients:", err)
+	}
+
+	var progressiveIncPayload []int
+	var mu sync.Mutex
+
+	invocationHandler := func(ctx context.Context, inv *wamp.Invocation) InvokeResult {
+
+		mu.Lock()
+		progressiveIncPayload = append(progressiveIncPayload, inv.Arguments[0].(int))
+		mu.Unlock()
+
+		if isInProgress, _ := inv.Details[wamp.OptProgress].(bool); !isInProgress {
+			var sum int64
+			for _, arg := range progressiveIncPayload {
+				n, ok := wamp.AsInt64(arg)
+				if ok {
+					sum += n
+				}
+			}
+			return InvokeResult{Args: wamp.List{sum}}
+		}
+
+		return InvokeResult{Err: wamp.InternalProgressiveOmitResult}
+	}
+
+	procName := "nexus.test.progproc"
+
+	// Register procedure
+	if err = callee.Register(procName, invocationHandler, nil); err != nil {
+		t.Fatal("Failed to register procedure:", err)
+	}
+
+	// Test calling the procedure.
+	callArgs := wamp.List{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+	ctx := context.Background()
+
+	callSends := 0
+	sendProgDataCb := func(ctx context.Context) (options wamp.Dict, args wamp.List, kwargs wamp.Dict, err error) {
+		options = wamp.Dict{}
+
+		if callSends == (len(callArgs) - 1) {
+			options[wamp.OptProgress] = false
+		} else {
+			options[wamp.OptProgress] = true
+		}
+
+		args = wamp.List{callArgs[callSends]}
+		callSends++
+
+		time.Sleep(100 * time.Millisecond)
```

Comment:
is the sleep necessary?

Reply:
well, yeah, seems no need. Removed it.
But then tests starts failing: payloads are not ordered. This means 
that something is not ordered... But i can't figure out what is it?

In `CallProgressive` the first one callback is fired and then `c.sess.Send` is called which in turn should just add a message to localpeer queue.
So all next `sendProg` calls that happening in the goroutine in loop should also just push a message to localpeer buffered channel....

Reply:
I'll try to play with the branch locally and see what's going on there.